### PR TITLE
Add i18n documentation

### PR DESCRIPTION
**Pull request by julianwachholz**
Initial draft for contribution guidelines on how to prepare Actual for i18n and l10n.

Please do add your feedback!

### CODE REVIEW

File: docs/contributing/i18n.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Internationalization
+
+You're interested in bringing Actual to a broader audience by translating and localizing the app into new languages? Great! This guide will show you how to get started. Adding translations can be a great first step into contributing to open source software.
```
Comment (RubenOlsen) [inline docs/contributing/i18n.md:3]: I understand the context, but the form is odd. 

Suggestion #1: _If you're interested in bringing..._
Suggestion #2: _Are you interested in bringing..._

File: docs/contributing/i18n.md
Comment (psybers) [inline docs/contributing/i18n.md:1]: This file is not in the menu, and thus no one will ever find it unless they guess the URL.

Be sure to also edit https://github.com/actualbudget/docs/blob/master/docs-sidebar.js

File: docs/contributing/i18n.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Current status
+
+Actual is currently being prepared for internationalization (i18n). This means that originally everything in the app was English, but we're slowly marking all texts for translation.
```
Comment (RubenOlsen) [inline docs/contributing/i18n.md:7]: - Consider using _initially,_ over _originally_
- Suggestion to make the second part of the paragraph a bit more clear: _This means that everything in the app was initially English, but we're slowly marking all texts for translation._

File: docs/contributing/i18n.md (hunk 1 above)
Comment (psybers) [inline docs/contributing/i18n.md:3]:
```suggestion
Are you interested in bringing Actual to a broader audience by translating and localizing the app into new languages? Great! This guide will show you how to get started. Adding translations can be a great first step into contributing to open-source software.
```

File: docs/contributing/i18n.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+At this time Actual **is not yet ready** to receive community translations.
```
Comment (RubenOlsen) [inline docs/contributing/i18n.md:9]: Suggestion: _At this time_ => _Currently_.

File: docs/contributing/i18n.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+## Marking for translation
+
+Developing translated software requires that every text shown to the user needs to pass through a translation function. Actual uses [i18next](https://www.i18next.com/) to achieve this goal.
```
Comment (RubenOlsen) [inline docs/contributing/i18n.md:13]: Consider removing: _needs to_ to make the sentence shorter.

File: docs/contributing/i18n.md (hunk 3 above)
Comment (psybers) [inline docs/contributing/i18n.md:9]: I'd recommend making this use the warning box. And possibly moving to the very top of the page.

```suggestion
:::warning

At this time Actual **is not yet ready** to receive community translations.

:::
```

File: docs/contributing/i18n.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+Strings in Actual are translated by their natural English language version. The project does not use artificial keys to select the appropriate translated
+text from the message catalogue.
```
Comment (psybers) [inline docs/contributing/i18n.md:16]: I still don't know if we are using American or British English for the docs.... but I feel like most of the other pages read like American English.

```suggestion
text from the message catalog.
```

File: docs/contributing/i18n.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+When preparing your translatable strings, please use complete units to translate. If there is a cohesive paragraph with multiple sentences, mark the entire paragraph as one translatable string. Do not split it into multiple texts.
+
```
Comment (RubenOlsen) [inline docs/contributing/i18n.md:18]: These two sentences should be augmented by examples of what one means by _artificial keys_ vs _natural English_. If one is deep into translating software this is a go-given, but if you have never translated any software before those sentences needs a little bit more explanation.

**Review by psybers — CHANGES_REQUESTED** [review]

**Review by RubenOlsen — CHANGES_REQUESTED** [review]

**Review by psybers — APPROVED** [review]

**Review by youngcw — APPROVED** [review]

**Pull request merged**